### PR TITLE
Polish sort & tex generate in py

### PR DESCRIPTION
**Pull request by ktab15**
Adding polish sorting (ICU) and file songs2tex with generating songbook in tex

### CODE REVIEW

File: src/latex/song2tex.py [hunk 1]
```diff
@@ -109,48 +115,83 @@ def parseDOM(root, linked=False):
...
-    trim_blocks = True,
-    lstrip_blocks = True,
-    autoescape = False,
-    loader = jinja2.FileSystemLoader(sys.path[0])
-)
-template = latex_jinja_env.get_template('song_template.tex')
-print(template.render(song=song))
+
+def s2t(path):
+    path = path
```
Comment (ptabor) [inline src/latex/song2tex.py:149]: this is a no-op

File: src/latex/song2tex.py [hunk 2]
```diff
@@ -109,48 +115,83 @@ def parseDOM(root, linked=False):
...
+
+
+def main():
+    if len(sys.argv) < 2:
+        print("Wymagana nazwa pliku xml", file=sys.stderr)
+        exit(1)
+    tree = etree.parse(sys.argv[1])
+    song = Song.parseDOM(tree.getroot())
+
+    latex_jinja_env = jinja2.Environment(
```
Comment (ptabor) [inline src/latex/song2tex.py:178]: Let's reuse song2tex method instead of duplicating the logic.

File: src/latex/songs2tex.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import os
```
Comment (ptabor) [inline src/latex/songs2tex.py:1]: Please describe in a comment above, what the code is doing.

File: src/latex/songs2tex.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            print(f"There is no such a file: {s}", file=sys.stderr)
+            exit(1)
+
+    collator = icu.Collator.createInstance(icu.Locale('pl_PL.UTF-8'))
+    list_title_file.sort(key=lambda x: collator.getSortKey(x[0]))
+
+    # I define head (template of header of tex document) and foot (template of footer of tex document)
+    # depending on the papersize which we want
+
+    if papersize != "a4":
```
Comment (ptabor) [inline src/latex/songs2tex.py:38]: no need for nested if... you can use logical operator or `in`

File: src/latex/songs2tex.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        foot = f"songbook_{papersize}_s.tex"
+    else:
+        head = f"single_{papersize}_p.tex"
+        foot = f"single_{papersize}_s.tex"
+
+    with open("src/formats/" + head) as f:
+        content = f.read()
+
+    if songbook:
+        p = content.find(":title:")
```
Comment (ptabor) [inline src/latex/songs2tex.py:54]: How about using: 
  https://www.programiz.com/python-programming/methods/string/replace

File: src/latex/songs2tex.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        s = content[:p] + title_of_songbook + content[p + 7:]
+        content = s
+
+    for song in list_title_file:
```
Comment (ptabor) [inline src/latex/songs2tex.py:58]: I think that such form could work and be a little bit more readable: 
```
for _,file in list_title_file:
        content += s2t.s2t(file)
```

From performance perspective I would consider: 

 content="\n\n".join(s2t.song2tex(file) for _, file in list_title_file);

**Review by ptabor — COMMENTED** [review]
Thank you. Just a bunch of minor fixes. But LGTM overall.

File: src/epub/create_epub.py [hunk 7]
```diff
@@ -140,9 +144,11 @@ def create_full_epub(src_of_songs, src, target_dir):
     create_template_epub(target_dir)
     path_out = os.path.join(target_dir, "epub", "OEBPS")
     cash.create_all_songs_html(src_of_songs, src, path_out)
-    create_content_opf(list_of_song(path_out), target_dir)
-    create_toc_ncx(list_of_song(path_out), target_dir)
-    create_toc_xhtml(list_of_song(path_out), target_dir)
+    los = list_of_song(path_out)
+    print(los)
```
[the remark this answers is not in the document]
Comment (ktab15) [reply src/epub/create_epub.py:148]: Done

File: src/epub/create_epub.py [hunk 8]
```diff
@@ -41,7 +42,10 @@ def list_of_song(path_out):
         if os.path.splitext(song)[1] == '.xhtml':
             path = "../../songs/" + os.path.splitext(song)[0]
             add_song(path, list_od_meta)
-    list_od_meta.sort(key=lambda x: x.title)
+    collator = icu.Collator.createInstance(icu.Locale('pl_PL.UTF-8'))
+    list_od_meta.sort(key=lambda x: collator.getSortKey(x.title))
+    print(list_od_meta[-10].title)
```
[the remark this answers is not in the document]
Comment (ktab15) [reply src/epub/create_epub.py:47]: Done

File: src/latex/song2tex.py [hunk 9]
```diff
@@ -109,48 +115,83 @@ def parseDOM(root, linked=False):
...
-    line_comment_prefix = '%#',
-    trim_blocks = True,
-    lstrip_blocks = True,
-    autoescape = False,
-    loader = jinja2.FileSystemLoader(sys.path[0])
-)
-template = latex_jinja_env.get_template('song_template.tex')
-print(template.render(song=song))
+
+def s2t(path):
```
[the remark this answers is not in the document]
Comment (ktab15) [reply src/latex/song2tex.py:148]: done

File: src/latex/song2tex.py (hunk 1 above)
Comment (ktab15) [reply src/latex/song2tex.py:149]: done

File: src/latex/songs2tex.py (hunk 3 above)
Comment (ktab15) [reply src/latex/songs2tex.py:1]: Done, check English understanding :p

File: src/latex/songs2tex.py (hunk 4 above)
Comment (ktab15) [reply src/latex/songs2tex.py:38]: Done

File: src/latex/song2tex.py (hunk 2 above)
Comment (ktab15) [reply src/latex/song2tex.py:178]: Done

**Review by ptabor — APPROVED** [review]
Awesome. Thank you for fixing.

**Pull request merged**